sda/cmd/api/api.go

```diff
@@ -234,10 +241,26 @@ func checkDB(db *database.SDAdb, timeout time.Duration) error {
 	return db.DB.PingContext(ctx)
 }
 
+func auditLog(token jwt.Token, c *gin.Context) {
+	if token == nil {
+		log.Warn("Empty JWT token instance")
+
+		return
+	}
+	if auditLogger != nil {
+		auditLogger.WithFields(log.Fields{
+			"audit": true,
+			"user":  token.Subject(),
+			"path":  c.Request.URL.Path,
+		}).Info("incoming audit event")
+	}
+}
+
```

Comment:
If you make the function more generic what should be logged can be decided at the time of calling and you don't needs any if statements in the function.

```suggestion
func auditLog(auditFields log.Fields) {
	auditFields["audit"] =  true
	auditLogger.WithFields(auditFields).Info("incoming audit event")
}

```

Then when it is called it will be like this:

```
if conf.API.auditLogger != nil {
	auditLog(log.Fields{
			"user":  token.Subject(),
			"path":  c.Request.URL.Path,
		})
}
```

And in the case of a JWT error we could log something like this:

```
if conf.API.auditLogger != nil {
	auditLog(log.Fields{
			"authentication error":  err.Error(),
			"path":  c.Request.URL.Path,
		})
}
```